metricflow/test/dataflow_plan_to_svg.py

```diff
@@ -28,35 +29,46 @@ def add_nodes_to_digraph(node: DagNodeT, dot: graphviz.Digraph) -> None:
 DagGraphT = TypeVar("DagGraphT", bound=MetricFlowDag)
 
 
-def display_graph_as_svg(
+def display_graph_if_requested(
     mf_test_session_state: MetricFlowTestSessionState,
     request: FixtureRequest,
     dag_graph: DagGraphT,
 ) -> None:
-    """Create and display the plan as an SVG, if configured to do so."""
+    """Create and display the plan as an SVG, if requested to do so."""
 
     if not mf_test_session_state.display_plans:
         return
+
     plan_svg_output_path_prefix = snapshot_path_prefix(
         request=request, snapshot_group=dag_graph.__class__.__name__, snapshot_id=dag_graph.dag_id
     )
 
-    dot = graphviz.Digraph(comment=dag_graph.dag_id, node_attr={"shape": "box", "fontname": "Courier"})
-    dot.format = "svg"
-
-    # Not quite correct if there are shared nodes.
-    for sink_node in dag_graph.sink_nodes:
-        add_nodes_to_digraph(sink_node, dot)
-
     # Create parent directory since it might not exist
     os.makedirs(os.path.dirname(plan_svg_output_path_prefix), exist_ok=True)
 
-    dot.format = "svg"
-
     if mf_test_session_state.plans_displayed >= mf_test_session_state.max_plans_displayed:
         raise RuntimeError(
-            f"Can't display plan - hit limit of " f"{mf_test_session_state.max_plans_displayed} plans displayed."
+            f"Can't display plan - hit limit of {mf_test_session_state.max_plans_displayed} plans displayed."
         )
-    # Don't use the .svg extension since it will add it.
-    dot.render(plan_svg_output_path_prefix, view=mf_test_session_state.display_plans, format="svg", cleanup=True)
+    _render_via_graphviz(dag_graph=dag_graph, file_path_without_svg_suffix=plan_svg_output_path_prefix)
     mf_test_session_state.plans_displayed += 1
+
+
+def display_dag_as_svg(dag_graph: DagGraphT, mf_config_dir: str) -> str:
+    """Create and display the plan as an SVG in the browser.
+
+    Returns the path where the SVG file was created within "mf_config_dir".
+    """
+    svg_dir = os.path.join(mf_config_dir, "generated_svg")
+    random_file_path = os.path.join(svg_dir, f"dag_{random_id()}")
+    _render_via_graphviz(dag_graph=dag_graph, file_path_without_svg_suffix=random_file_path)
+    return random_file_path + ".svg"
+
+
+def _render_via_graphviz(dag_graph: DagGraphT, file_path_without_svg_suffix: str) -> None:
+    dot = graphviz.Digraph(comment=dag_graph.dag_id, node_attr={"shape": "box", "fontname": "Courier"})
+    # Not quite correct if there are shared nodes.
+    for sink_node in dag_graph.sink_nodes:
+        add_nodes_to_digraph(sink_node, dot)
+    dot.format = "svg"
```

Comment:
Could we make the format configurable here? I find SVG kind of annoying to view.

Reply:
What are you running into? When using the option, a browser window pops up automatically and renders without any issues for me.

Reply:
Oh got it. I haven't tried running the command. This default behavior should make viewing it easy.